apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -338,9 +338,13 @@ func (p *ProvisionerAPI) machineLXDProfileNames(m *state.Machine, env environs.E
 			continue
 		}
 		pName := lxdprofile.Name(p.m.Name(), app.Name(), ch.Revision())
+		// Lock here, we get a new env for every call to ProvisioningInfo().
```

Comment:
0        down                  pending        bionic      cannot write lxd profiles: Error inserting juju-default-lxd-profile-0 into database: The profile already exists

Reply:
Bumped into this on the acceptance tests when overloading the machines. Can confirm this allows us to pass the acceptance tests now.